Use Ninja as generator in HPX package

msimberg opened the pull request:
Due to limitations in Make or CMake's Makefile generation, using Make as the generator significantly limits the parallelism available for building HPX. CMake+Ninja does a much better job of exposing parallelism, and as such I think it's worth changing the default generator.

I couldn't tell from other packages if this needs an additional `depends_on('ninja', type='build')` as well. It works without it so I assume it's fine like this, but please let me know if it's preferable for some reason to have it there.

msimberg committed:
Use Ninja as generator in HPX package

albestro commented:
I gave it a test by patching my hpx package and it does not work: it complains as expected because I don't have ninja installed on my macbook.

I added the build dependency, and it is building as expected.

So, I confirm that we have to add it.

msimberg commented:
> I gave it a test by patching my hpx package and it does not work: it complains as expected because I don't have ninja installed on my macbook.
> 
> I added the build dependency, and it is building as expected.
> 
> So, I confirm that we have to add it.

Ah, thanks @albestro for checking! In that case it just happened to work for me because I have ninja in my PATH. I added it here as well.
